component/otelcol/processor/transform/transform.go

```diff
@@ -0,0 +1,173 @@
+			fact := transformprocessor.NewFactory()
+			return processor.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+type ContextID string
+
+const (
+	Resource  ContextID = "resource"
+	Scope     ContextID = "scope"
+	Span      ContextID = "span"
+	SpanEvent ContextID = "spanevent"
+	Metric    ContextID = "metric"
+	DataPoint ContextID = "datapoint"
+	Log       ContextID = "log"
+)
+
+func (c *ContextID) UnmarshalText(text []byte) error {
+	str := ContextID(strings.ToLower(string(text)))
+	switch str {
+	case Resource, Scope, Span, SpanEvent, Metric, DataPoint, Log:
+		*c = str
+		return nil
+	default:
+		return fmt.Errorf("unknown context %v", str)
+	}
+}
+
+type contextStatementsSlice []contextStatements
+
+type contextStatements struct {
+	Context    ContextID `river:"context,attr"`
+	Statements []string  `river:"statements,attr"`
+}
+
+// Arguments configures the otelcol.processor.transform component.
+type Arguments struct {
+	// ErrorMode determines how the processor reacts to errors that occur while processing a statement.
+	ErrorMode        ottl.ErrorMode         `river:"error_mode,attr,optional"`
+	TraceStatements  contextStatementsSlice `river:"trace_statements,block,optional"`
+	MetricStatements contextStatementsSlice `river:"metric_statements,block,optional"`
+	LogStatements    contextStatementsSlice `river:"log_statements,block,optional"`
+
+	// Output configures where to send processed data. Required.
+	Output *otelcol.ConsumerArguments `river:"output,block"`
+}
+
+var (
+	_ processor.Arguments = Arguments{}
+)
+
+// DefaultArguments holds default settings for Arguments.
+var DefaultArguments = Arguments{
+	ErrorMode: ottl.PropagateError,
+}
+
+// SetToDefault implements river.Defaulter.
+func (args *Arguments) SetToDefault() {
+	*args = DefaultArguments
+}
+
+// Validate implements river.Validator.
+func (args *Arguments) Validate() error {
+	otelArgs, err := args.convertImpl()
+	if err != nil {
+		return err
+	}
+	return otelArgs.Validate()
+}
+
+func (stmts *contextStatementsSlice) convert() []interface{} {
+	if stmts == nil {
+		return nil
+	}
+
+	res := make([]interface{}, 0, len(*stmts))
+
+	if len(*stmts) == 0 {
+		return res
+	}
+
+	for _, stmt := range *stmts {
+		res = append(res, stmt.convert())
+	}
+	return res
+}
+
+func (args *contextStatements) convert() map[string]interface{} {
+	if args == nil {
+		return nil
+	}
+
+	return map[string]interface{}{
+		"context":    args.Context,
+		"statements": args.Statements,
+	}
+}
+
+// Convert implements processor.Arguments.
+func (args Arguments) Convert() (otelcomponent.Config, error) {
+	return args.convertImpl()
+}
+
+// convertImpl is a helper function which returns the real type of the config,
+// instead of the otelcomponent.Config interface.
+func (args Arguments) convertImpl() (*transformprocessor.Config, error) {
+	input := make(map[string]interface{})
+
+	input["error_mode"] = args.ErrorMode
+
+	if len(args.TraceStatements) > 0 {
```

Comment:
Could this be simplified by removing the len check? Your checking for nil in the convert itself.

Reply:
Sometimes in the Collector config the behaviour is different even if none of the underlying config is set. E.g. I think this will enable both grpc and http receivers:
```
receivers:
  otlp:
    protocols:
      grpc:
      http:
```
But I think this will enable only http:
```
receivers:
  otlp:
    protocols:
      http:
```
In the case of this component, I think the behaviour would be the same, but I'd still want to refrain from setting this unless we want this field to be used with its default values.